exercises/kindergarten-garden/src/example/java/KindergartenGarden.java

```diff
@@ -0,0 +1,44 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+
+
+public class KindergartenGarden {
+    private String[] students;
+    private String garden;
+    private final int PLANTS_PER_STUDENT_PER_ROW = 2;
+    private final nt NEW_ROW_LOCATION;
+
+    private final static String[] DEFAULT_STUDENTS = {
+            "Alice", "Bob", "Charlie", "David",
+            "Eve", "Fred", "Ginny", "Harriet",
+            "Ileana", "Joseph", "Kincaid", "Larry"
+    };
+
+    public KindergartenGarden(String garden, String[] students) {
+        this.garden = garden;
+        Arrays.sort(students);
+        this.students = students;
+        NEW_ROW_LOCATION = garden.indexOf('\n') + 1;
+    }
+
+    public KindergartenGarden(String garden) {
+        this(garden, DEFAULT_STUDENTS);
+    }
+
+    public Plant[] getPlantsOfStudent(String student) {
+        List<Plant> plants = new ArrayList<Plant>();
+        int studentPlantsIndex = Arrays.binarySearch(students, student) * PLANTS_PER_STUDENT_PER_ROW;
+
+        for (int i = studentPlantsIndex; i < studentPlantsIndex + PLANTS_PER_STUDENT_PER_ROW; i++) {
```

Comment:
These `for` loops look much neater now! :D Come to think of it, as you're only adding four things, it might be easier to just have four `plants.add` statements? 
```
plants.add(Plant.getPlant(garden.chatAt(studentPlantsIndex)));
plants.add(Plant.getPlant(garden.charAt(studentPlantsIndex + 1)));
plants.add(Plant.getPlant(garden.charAt(studentPlantsIndex + NEW_ROW_LOCATION)));
plants.add(Plant.getPlant(garden.charAt(studentPlantsIndex + NEW_ROW_LOCATION + 1)));
```
It doesn't really matter though. Both make it pretty clear what you're doing :)